src/service/viewer-impl.js

```diff
@@ -403,6 +403,9 @@ export class Viewer {
       this.hasBeenVisible_ = true;
       this.whenFirstVisibleResolve_();
     }
+    if (!this.visibilityObservable_) {
```

Comment:
This is called from the constructor, so it isn't really lazy. Probably better to make it eager, then.

Laziness primarily makes sense when the instantiation is only rarely done subsequently.